### PR TITLE
fix: Add CI test build step and update build libs

### PR DESCRIPTION
**Pull request by NathanFarmer**
- Update cjs build target from `ES6` to `ESNEXT`
- Add CI step to test build

### CODE REVIEW

File: tsconfig.cjs.json [hunk 1]
```diff
@@ -2,10 +2,10 @@
     "extends": "./tsconfig.production.json",
     "compilerOptions": {
         "lib": [
-            "ES6",
+            "ESNEXT",
             "DOM"
         ],
-        "target": "ES6",
+        "target": "ESNEXT",
```
Comment (DrewHoo) [inline tsconfig.cjs.json:8]: Why change target from ES6?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -39,10 +39,20 @@ jobs:
...
             ${{ runner.os }}-pnpm-store-
-
+          
       - name: Install dependencies
         run: pnpm install --frozen-lockfile
-      
+
+      - name: Dry-run of build
+        shell: bash
+        run: |
```
Comment (DrewHoo) [inline .github/workflows/tests.yaml:48]: I think there's a cli arg that specifies a dry run here. Also curious why the pipe operator?

**Review by DrewHoo — CHANGES_REQUESTED** [review]

File: tsconfig.cjs.json (hunk 1 above)
Comment (NathanFarmer) [reply tsconfig.cjs.json]: My mistake

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (NathanFarmer) [reply .github/workflows/tests.yaml]: @DrewHoo I looked for dry run, but the only thing I could find was `tsc --build --dry` and that doesn't fail like `tsc -p` does 🤷🏻

Comment (DrewHoo) [reply .github/workflows/tests.yaml]: Did you look at https://semantic-release.gitbook.io/semantic-release/usage/configuration#dryrun

If this is complicated, i think it's fine to defer & tackle later in another PR

Comment (NathanFarmer) [reply .github/workflows/tests.yaml]: I tried `npx semantic-release --dry-run`, but it doesn't fail that way either.

**Review by DrewHoo — APPROVED** [review]

**Pull request merged**